src/http/api/views/account.follows.view.ts

```diff
@@ -179,11 +226,90 @@ export class AccountFollowsView {
             return error('error-getting-follows');
         }
 
-        if (!page) {
-            return error('no-page-found');
-        }
+        // Handling non-paginated follow lists
+        if (!page || Number(next)) {
+            const follows =
+                type === 'following'
+                    ? await actor.getFollowing()
+                    : await actor.getFollowers();
 
-        const accounts: AccountInfo[] = [];
+            if (!follows) {
+                return error('error-getting-follows');
+            }
+
+            console.log('Follows: ', follows);
+
+            const pageSize = 15;
+            const pageNumber = next ? Number.parseInt(next, 10) : 1;
+            const startIndex = (pageNumber - 1) * pageSize;
+
+            const pageUrls = follows.itemIds.slice(
+                startIndex,
+                startIndex + pageSize,
+            );
+
+            for await (const item of pageUrls) {
+                try {
```

Comment:
This logic is (almost?) exactly the same as the loop below - are we able to deduplicate this - the method is getting pretty long!

Reply:
Done here https://github.com/TryGhost/ActivityPub/pull/557/commits/b0204ee194bee737dffdc75fa473d1cd9b730f3a